wandb/data_types.py

```diff
@@ -6,2657 +6,94 @@
-            img_width, img_height = image._image.size
-            return img_width == width and img_height == height
-
-        sizes_match = all(size_equals_image(img) for img in images)
-        if not sizes_match:
-            logging.warning(
-                "Images sizes do not match. This will causes images to be display incorrectly in the UI."
-            )
-
-        meta = {
-            "_type": "images/separated",
-            "width": width,
-            "height": height,
-            "format": format,
-            "count": num_images_to_log,
-        }
-
-        captions = Image.all_captions(images)
-
-        if captions:
-            meta["captions"] = captions
-
-        all_masks = Image.all_masks(images, run, key, step)
-
-        if all_masks:
-            meta["all_masks"] = all_masks
-
-        all_boxes = Image.all_boxes(images, run, key, step)
-
-        if all_boxes:
-            meta["all_boxes"] = all_boxes
-
-        return meta
-
-    @classmethod
-    def all_masks(cls, images, run, run_key, step):
-        all_mask_groups = []
-        for image in images:
-            if image._masks:
-                mask_group = {}
-                for k in image._masks:
-                    mask = image._masks[k]
-                    mask_group[k] = mask.to_json(run)
-                all_mask_groups.append(mask_group)
-            else:
-                all_mask_groups.append(None)
-        if all_mask_groups and not all(x is None for x in all_mask_groups):
-            return all_mask_groups
-        else:
-            return False
-
-    @classmethod
-    def all_boxes(cls, images, run, run_key, step):
-        all_box_groups = []
-        for image in images:
-            if image._boxes:
-                box_group = {}
-                for k in image._boxes:
-                    box = image._boxes[k]
-                    box_group[k] = box.to_json(run)
-                all_box_groups.append(box_group)
-            else:
-                all_box_groups.append(None)
-        if all_box_groups and not all(x is None for x in all_box_groups):
-            return all_box_groups
-        else:
-            return False
-
-    @classmethod
-    def all_captions(cls, images):
-        if images[0]._caption != None:
-            return [i._caption for i in images]
-        else:
-            return False
-
-    def __ne__(self, other):
-        return not self.__eq__(other)
-
-    def __eq__(self, other):
-        return (
-            self._grouping == other._grouping
-            and self._caption == other._caption
-            and self._width == other._width
-            and self._height == other._height
-            and self._image == other._image
-            and self._classes == other._classes
-        )
-
-
-# Allows encoding of arbitrary JSON structures
-# as a file
+# This file is maintained so that users can access the data_types module
+# as if it is a top-level module, rather than reaching into the sdk folder:
```

Comment:
why does anyone need to do this, currently I thought we exported all these objects at the top level of the module (not ideal).

Reply:
I just didn't want to break existing scripts. Anything we have control over can easily change from `wandb.data_types` to `wandb.sdk.data_types`, but i thought maybe users would have scripts with `wandb.data_types` somewhere. It would be much better to remove this (: